tests/integration_tests/test_chainermn.py

```diff
@@ -300,7 +302,11 @@ def test_suggest_float_with_step(storage_mode: str, comm: CommunicatorBase) -> N
                 x2 = mn_trial.suggest_float("x", low, high, step=step)
                 assert x1 == x2
 
-                mn_trial.suggest_float("x", low, high)
+                if comm.rank == 0:
+                    with pytest.warns(RuntimeWarning):
+                        mn_trial.suggest_float("x", low, high)
+                else:
+                    mn_trial.suggest_float("x", low, high)
```

Comment:
Only rank 0 node shows `RuntimeWarning`, so I use the if branch.

Reply:
Could you tell me how do you select `warnings.catch_warnings` and `pytest.warns`. If we use `warnings.catch_warnings`, we may not need the condition branch.

Reply:
Since the other `test_suggest_*` methods use `with pytest.raises(ValueError)` such as https://github.com/optuna/optuna/blob/19a7fa5a48734978affecd54d501f61a8cfb4b94/tests/integration_tests/test_chainermn.py#L332-L334, I use `with pytest.warns(RuntimeWarning)` for consistency. But I suppose it doesn't look consistent with other methods, so I'm happy to use `warnings.catch_warnings` instead of the current `with pytest.raises(ValueError)`.